### PR TITLE
Feature/v2.4.0

### PR DESCRIPTION
**Pull request by metinavc1**

### CODE REVIEW

File: ramenpos/RamenPos/Logic/RamenApp+SPIConnectionDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        
+        if let secrets = secrets {
+            SPILogMsg("\n\n")
+            SPILogMsg("# ----------- I GOT NEW SECRETS -----------")
+            SPILogMsg("# ---------- PERSIST THEM SAFELY ----------")
+            SPILogMsg("# \(String(describing: secrets.encKey)):\(String(describing: secrets.hmacKey))")
+            SPILogMsg("# -----------------------------------------")
+            
+            // In prod store them in the key chain
+            settings.encriptionKey = secrets.encKey!
```
Comment (chrishulbert) [inline ramenpos/RamenPos/Logic/RamenApp+SPIConnectionDelegate.swift]: Probably best to start off a new project with as few `!` as possible, they quickly get out of control if you let them, and you can end up crashier than the equivalent objc. If let or guard are good options.

File: ramenpos/RamenPos/Logic/Settings/SettingsProvider.swift [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        get { return readSettingsFrom(key: .hmacKey) as? String }
+        set { setSettingsForKey(key: .hmacKey, value: newValue) }
+    }
+    
+    var customerReceiptFromEftpos: Bool? {
+        get { return readSettingsFrom(key: .customerReceipt) as? Bool }
+        set { setSettingsForKey(key: .customerReceipt, value: newValue) }
+    }
+    
+    var customerSignatureromEftpos: Bool? {
```
Comment (chrishulbert) [inline ramenpos/RamenPos/Logic/Settings/SettingsProvider.swift]: Missing an 'f' :) 
```suggestion
    var customerSignatureFromEftpos: Bool? {
```

**Review by chrishulbert — APPROVED** [review]
Looks fine, i think the only things i noticed aren't dealbreakers, approved 🍺

File: ramenpos/RamenPos/Logic/RamenApp+SPIConnectionDelegate.swift (hunk 1 above)
Comment (metinavc1) [reply ramenpos/RamenPos/Logic/RamenApp+SPIConnectionDelegate.swift]: Removed '!'

File: ramenpos/RamenPos/Logic/Settings/SettingsProvider.swift (hunk 2 above)
Comment (metinavc1) [reply ramenpos/RamenPos/Logic/Settings/SettingsProvider.swift]: Fixed

**Review by chrishulbert — APPROVED** [review]

**Pull request merged**